pkg/metadata/compiler.go

```diff
@@ -38,17 +53,34 @@ func Compiler() *Provider {
 		if err != nil {
 			return nil, fmt.Errorf("failed to get path for process %d: %w", pid, err)
 		}
-		file, err := elf.Open(path)
+		elf, err := elf.Open(path)
 		if err != nil {
 			return nil, fmt.Errorf("failed to open ELF file for process %d: %w", pid, err)
 		}
-		defer file.Close()
+		defer elf.Close()
 
-		return model.LabelSet{
+		buildID, err := buildid.BuildID(path)
+		if err != nil {
+			return nil, fmt.Errorf("buildID failed")
+		}
+
+		value, ok := c.GetIfPresent(buildID)
+		if ok {
+			cachedLabels, ok := value.(model.LabelSet)
+			if !ok {
+				panic("The buildID cache contained the wrong type. This should never happen")
```

Comment:
Let's not panic. It would be better to return an error and delete the wrongly cached items.

Reply:
For conditions that should absolutely never happen, I like panicking as it can ensure that both readers of the code understand that this is a "hard" assertion, and also ensure that if something changes in the code, later on, we have more chances of catching it. 

What do you think?

Reply:
I agree with Javier. I think it's good to panic here instead of returning an error.

Reply:
Let's agree to disagree how we approach to panics :) It's fine for here.